pyresample/geometry.py

```diff
@@ -1820,9 +1861,9 @@ def create_areas_def_legacy(self):
     def __eq__(self, other):
         """Test for equality."""
         try:
-            return ((self.crs == other.crs) and
-                    (self.shape == other.shape) and
-                    (np.allclose(self.area_extent, other.area_extent)))
+            return ((np.allclose(self.area_extent, other.area_extent)) and
+                    (self.crs == other.crs) and
+                    (self.shape == other.shape))
```

Comment:
The change in order ensures the equality check raises `AttributeError` for `SwathDefinition`s. This PR introduced the `crs` kwarg to them so the lazy evaluation didn't go until the `area_extent` check. `SwathDefinition` lacks the extents so they'll raise the error.

Reply:
But the order change doesn't make this work better, does it? It just makes it so the error happens earlier than it would have before? The `.crs` failure would have passed or not passed and then gotten the attribute error for the extents afterward, right?

Reply:
No. The if statements are lazy and thus the order matters in try/except cases:
```bash
$ python -c 'if False and asdasd: pass'
$ python -c 'if asdasd and False: pass'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'asdasd' is not defined
```

Reply:
Sure but in this case the old comparison was CRS -> shape -> extent so it would fail on CRS first with an AttributeError, right? With that old order and the new `SwathDefinition.crs` it would compare the CRS and either return False if CRS didn't match or go on to shape and extent comparison which would raise an AttributeError again but this time for extent. In either ordering don't we get an AttributeError eventually? Or are you guys trying to avoid the unnecessary .crs comparison if it is a swath<->area comparison?

Reply:
> Or are you guys trying to avoid the unnecessary .crs comparison if it is a swath<->area comparison?

yes